src/session.rs

```diff
@@ -1657,6 +1657,10 @@ impl Session {
 
         match &storage {
             FileStorage::Single(path) if nlayers > 1 => {
+                if let FileStorage::Single(mut pathbuf) = storage.clone() {
+                    pathbuf.pop();
+                    std::fs::create_dir_all(pathbuf.as_path()).expect("Error while creating directories");
```

Comment:
Instead of blowing up if there's an error, we should return an error from the function. You can just do:
```
std::fs::create_dir_all(pathbuf.as_path())?;
```

Reply:
I'm also wondering: why do we cloen the storae, instead of just cloning `path` from line 1659?

Reply:
> std::fs::create_dir_all(pathbuf.as_path())?;
Haven't know about that syntax.
> I'm also wondering: why do we clone the storage, instead of just cloning path from line 1659?
Sure thing.Haven't saw that.